dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/database/HibernateDatabaseInfoProvider.java

```diff
@@ -90,16 +96,40 @@ public void init()
 
         boolean spatialSupport = false;
 
-        // Check if postgis is installed, fail startup if not
+        boolean trigramSupport = false;
+
+        boolean btreeGinSupport = false;
 
         if ( !SystemUtils.isTestRun( environment.getActiveProfiles() ) )
         {
+            // Check if postgis extension is installed, fail startup if not
             spatialSupport = isSpatialSupport();
 
             if ( !spatialSupport )
             {
-                log.error( POSTGIS_MISSING_ERROR );
-                throw new IllegalStateException( POSTGIS_MISSING_ERROR );
+                log.error( String.format( EXTENSION_MISSING_ERROR, POSTGIS_EXTENSION, POSTGIS_EXTENSION ) );
+                throw new IllegalStateException(
+                    String.format( EXTENSION_MISSING_ERROR, POSTGIS_EXTENSION, POSTGIS_EXTENSION ) );
```

Comment:
Why do we now say that all the extensions are missing even if we only check one of them? Maybe I don't understand the code 😋 Hope we are not telling the admins to create more extensions than they need to.

Reply:
The message specifies the exact extension that is missing. So in of the if clauses, specific extension are checked and corresponding error messages are formed and sent. In this specific case "POSTGIS_EXTENSION" is part of the error message. In the remaining clauses, its different.

Reply:
ah, sorry I misread the code then 🤦🏻 😂